### ISSUE TITLE
"Mark of the Web" prevents Windows app from running when distributed as a ZIP file

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1890]
### Describe the bug

When a ZIP-packaged app is extracted and runs, a stacktrace is shown when loading DLLs fails (as seen in https://github.com/beeware/briefcase/discussions/1881).

This first happens when pythonnet tries to load its Python runtime DLL (Python.Runtime.dll) and then for the WebView DLL from toga-winforms.

The underlying issue is Windows' [Mark of the Web](https://learn.microsoft.com/en-us/deployoffice/security/internet-macros-blocked#additional-information-about-mark-of-the-web) (MotW) functionality that prevents running foreign code while the file is marked.

### Steps to reproduce

Package an app for Windows using ZIP. Extract and run that app on another machine.

### Expected behavior

The app should successfully run.

The issue can be worked around by manually removing the MotW from all DLLs in the app (via selecting "Unblock" in the DLL file's properties).

To bypass the MotW check automatically, we can tell the .NET runtime to [allow loading](https://learn.microsoft.com/en-us/previous-versions/dotnet/netframework-4.0/dd409252(v=vs.100)) remote sources.

This can be accomplished via a `<App Formal Name>.exe.config` file alongside the app's stub exe:
```xml
<configuration>
   <runtime>
      <loadFromRemoteSources enabled="true"/>
   </runtime>
</configuration>
```

Since this is not needed for MSI packages, it may be best to add this file only when packaging as a ZIP.

### Environment

- Operating System: Windows 11
- Python version: 3.11
- Software versions:
  - Briefcase: `0.3.20.dev15+g93ee7e9f`

**Comment by freakboy3742** [comment]
Given how simple the file is, it should be possible to inject this directly into the zip file at time of packaging without ever writing a file to disk.

However, before we do that, I'd like to understand what else this file is used for. If there's any possibility we might want to use it for any other app configuration, we might want to add it earlier in the build process.

**Comment by freakboy3742** [comment]
@saisharan0103 What you've described sounds like a fairly standard approach - build the feature as the ticket describes, then add a test. If you'd like to tackle this problem, you're welcome to take a look at our [contribution guide](https://briefcase.beeware.org/en/stable/how-to/contribute/), and develop a pull request.
